### DISCUSSION TITLE
Splits and equalizing on splitting

### DISCUSSION THREAD

**Discussion by jonathanj** [ghostty-org/ghostty#2947] [Feature Requests, Ideas] [answered]
In https://github.com/ghostty-org/ghostty/issues/831 the `equalize-splits` command was added, I'd like to suggest going one step further: Have an option to automatically equalize splits after splitting. iTerm 2 splits behave this way by default.

There are some quirks to equalizing though, for example if I do: split right, split right, split bottom, split right:

<img width="1713" alt="image" src="https://github.com/user-attachments/assets/ea0dc1e8-1105-4754-a956-68e080315c9c" />

If I equalize splits now, I would expect that my right splits each become 1/3 wide, and my bottom split becomes 1/2 high, however that's not quite what happens:

<img width="1713" alt="image" src="https://github.com/user-attachments/assets/773dea1d-7587-4d6a-a193-27527a25ea19" />

The right splits are not divided equally into thirds, and the bottom split is not divided equally into halves. I'm not sure I can understand what happened, but it wasn't what I was expecting.

For reference, this is what my sequence of splits produces in iTerm 2:

<img width="1714" alt="image" src="https://github.com/user-attachments/assets/3c367c26-2804-4f51-9c8a-6862b02a63df" />

**Comment by jonathanj** [comment]
FYI, if I change my sequence in ghostty, it will produce the same thing as iTerm 2: split right, split right, equalize splits, split bottom, split, right

**Comment by JakeWharton** [comment, marked as answer]
Answered in #3839, resulted in https://github.com/ghostty-org/ghostty/issues/2227#issuecomment-2564602539
